server/TMTServer.go

```diff
@@ -262,20 +262,22 @@ func runKMeans(positionMap map[uuid.UUID]infra.PositionVector, numClusters int)
 func (tserv *TMTServer) moveAgents() {
 	for _, agent := range tserv.GetAgentMap() {
 		agentPos := agent.GetPosition()
-		moveX, moveY := tserv.grid.GetValidMove(agentPos.X, agentPos.Y)
-		targetPos, posExists := agent.GetTargetPosition()
+		moveX, moveY := tserv.grid.GetValidMove(agentPos.X, agentPos.Y) // Gets random valid move, or same position of none exist
+		nextMove := infra.PositionVector{
+			X: moveX,
+			Y: moveY,
+		}
```

Comment:
You're potentially overwriting (and redundantly creating) an object at this point. It's better to only create where necessary (i.e., where the values X and Y are known)

I prefer my implementation. 

Remember, this is a grid world.